config/authsources.php

```diff
@@ -41,20 +41,38 @@
 
 $defaultspentityid = "$baseurl/auth/saml2/sp/metadata.php";
 
+// Process requested attributes.
+$attributes = [];
+$attributesrequired = [];
+
+foreach (explode(PHP_EOL, $saml2auth->config->requestedattributes) as $attr) {
+    $attr = trim($attr);
+    if (empty($attr)) {
+        continue;
+    }
+    if (substr($attr, -2, 2) === ' *') {
+        $attr = substr($attr, 0, -2);
+        $attributesrequired[] = $attr;
+        $attributes[] = $attr;
+    } else {
+        $attributes[] = $attr;
+    }
```

Comment:
Adding comments here should be useful. For example `' *'` is checked because that's the convention used for required fields.

Also some further simplification can be done.

```suggestion
    if (empty($attr)) {
        continue;
    }
    if (substr($attr, -2, 2) === ' *') {
        $attr = substr($attr, 0, -2);
        $attributesrequired[] = $attr;
    }
    
    $attributes[] = $attr;
```

Reply:
Is there a situation where we need to trim the attribute before further processing it? Or should it already be in a good shape. I am leaning towards the later, and thinking of simplifying the block even further into something like:

```
foreach (explode(PHP_EOL, $saml2auth->config->requestedattributes) as $attr) {
    // Match and add required attributes.
    if (substr($attr, -2, 2) === ' *') {
        $attr = substr($attr, 0, -2);
        $attributesrequired[] = $attr;
    }

    $attributes[] = $attr;
}
```